api/src/main/java/org/openmrs/module/appointments/dao/impl/AppointmentDaoImpl.java

```diff
@@ -123,14 +127,40 @@ public Appointment getAppointmentByUuid(String uuid) {
     @Override
     public List<Appointment> search(AppointmentSearchRequest appointmentSearchRequest) {
         Criteria criteria = sessionFactory.getCurrentSession().createCriteria(Appointment.class);
+
         criteria.add(Restrictions.eq("voided", false));
-        Date maxEndDate = appointmentSearchRequest.getEndDate();
-        Date startDate = appointmentSearchRequest.getStartDate();
-        if(maxEndDate!=null && startDate!=null)
-            criteria.add(Restrictions.between("startDateTime", startDate, maxEndDate));
+        criteria.addOrder(Order.asc("startDateTime"));
+        setDateCriteria(appointmentSearchRequest, criteria);
+        setPatientCriteria(appointmentSearchRequest, criteria);
+        setLimitCriteria(appointmentSearchRequest, criteria);
+
         return criteria.list();
     }
 
+    private void setPatientCriteria(AppointmentSearchRequest appointmentSearchRequest, Criteria criteria) {
+        if (StringUtils.isNotEmpty(appointmentSearchRequest.getPatientUuid())) {
+            criteria.createAlias("patient", "patient");
+            criteria.add(Restrictions.eq("patient.uuid", appointmentSearchRequest.getPatientUuid()));
+        }
+    }
+
+    private void setDateCriteria(AppointmentSearchRequest appointmentSearchRequest, Criteria criteria) {
+        if (appointmentSearchRequest.getStartDate() != null) {
+            criteria.add(Restrictions.ge("startDateTime", appointmentSearchRequest.getStartDate()));
+        }
+        if (appointmentSearchRequest.getEndDate() != null) {
+            criteria.add(Restrictions.le("startDateTime", appointmentSearchRequest.getEndDate()));
```

Comment:
shouldn't this be "endDateTime"?

Reply:
No.
The appointment startDate should be between the appointmentSearchRequest start and end dates.
Hence althought the check in on appointmentSearchRequest endDate , the criteria is set on appointment startDate